src/Domain.php

```diff
@@ -651,15 +691,63 @@ public function withoutLabel(int $key, int ...$keys): self
         }
 
         if ([] === $labels) {
-            return new self();
+            return new self(null, null, ...$this->getIDNAOptions());
         }
 
         $domain = implode('.', array_reverse($labels));
         $psContent = $this->publicSuffix->getContent();
         if (null === $psContent || '.'.$psContent !== substr($domain, - strlen($psContent) - 1)) {
-            return new self($domain);
+            return new self($domain, null, ...$this->getIDNAOptions());
         }
 
-        return new self($domain, $this->publicSuffix);
+        return new self($domain, $this->publicSuffix, ...$this->getIDNAOptions());
+    }
+    
+    /**
+     * @return array
+     */
+    public function getIDNAOptions(): array
```

Comment:
Having both type of methods seems overkill why not:

- just expose `getIDNAOptions`;
- or `getAsciiIDNAOption` and `getUnicodeIDNAOption`;

but not both...